### PR TITLE
Add preact plugin

### PR DESCRIPTION
**Pull request by just-be-dev**
This ports over the next plugin as was [requested](https://github.com/zeit/next-plugins/pull/487#issuecomment-510215560) by @timneutkens in https://github.com/zeit/next-plugins/pull/487

### CODE REVIEW

File: packages/next-preact/readme.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Next.js + preact
+
+Use [preact](https://preactjs.com/) with [Next.js](https://github.com/zeit/next.js)
+
+## Installation
+
+```
+npm install --save @next/preact preact
```
Comment (developit) [inline packages/next-preact/readme.md]:
```suggestion
npm install --save @next/preact preact@next
```

(lol)

**Review by developit — DISMISSED** [review]

File: packages/next-preact/readme.md (hunk 1 above)
Comment (zephraph) [reply packages/next-preact/readme.md]: Great catch! Thanks.

**Comment by tezhub** [comment]
Hi, can we use this plugin without `server.js` i.e in `serverless` target? It would be nice have this info in the Readme.

**Comment by zephraph** [comment]
Yes, I believe so. You just need to wrap your next config with `withPreact` as it says under the server section. I'm open to suggestions of how the readme could be clarified.

**Comment by timneutkens** [comment]
I think we should find a solution that doesn't require the custom server anymore. Maybe we don't really need a plugin for preact support as it's relatively simple to support, wdyt @developit?

**Comment by developit** [comment]
I'm not sure why module-alias is necessary in the first place here. Couldn't react be not marked as external and the Webpack resolve aliases would bundle Preact in its place, like they do on the client?

**Comment by zephraph** [comment]
@developit I took your suggestion and remove those parts. Did some brief testing and it seems to work fine... but I'm no next expert, so if folks could double check me that'd be grand. 

Let me know if there's anything else I need to do.

**Comment by zephraph** [comment]
The bin still prompts the user to install `react` and `react-dom`. I think we could likely update that to check if `@next/preact` is installed and if so just make sure that `preact` is installed.

**Comment by zephraph** [comment]
I'll check what's up w/ the tests tomorrow.

**Comment by zephraph** [comment]
@developit does this update look right?

**Comment by timneutkens** [comment]
Hey sorry for the delay in getting back to your. We've been discussing another avenue where the preact support would be part of the core and all you'd have to do is install preact, hence why I haven't merged this PR yet.

**Comment by zephraph** [comment]
Sounds good. I'll close for now then.

**Pull request closed without merging**